hil/ext/switches/n3000.py

```diff
@@ -104,16 +103,13 @@ def _sendline(self, line):
     @staticmethod
     def connect(switch):
         """connect to the switch and log in"""
-        console = pexpect.spawn(
-            'ssh ' + switch.username + '@' + switch.hostname)
+        console = _console.login(switch)
 
-        console.expect('password: ')
-        console.sendline(switch.password)
-        console.expect('>')
-        console.sendline('enable')
-
-        logger.debug('Logged in to switch %r', switch)
+        # send a new line so that we can "expect" a prompt again if we already
```

Comment:
login using pubkey will most likely work on the n3000 since the `_console.login` is pretty generic. But because I haven't tested it, I am not documenting/claiming that it will work.

Reply:
`_console.login` specifies that it does pubkey auth, so a reader of this code might naturally assume that it works unless stated otherwise. Ideally we'd just test it so this could be a moot point, but otherwise I'd at least like a comment mentioning that it's untested.